RELEASES.md

```diff
@@ -0,0 +1,41 @@
+# Releases Guide
+
+Here, we will briefly understand the release plan and versioning steps. 
+This will cover the release process, major and minor releases, backporting the PR and manual release steps.
+
+If there is something that you require or this document leaves out, please
+reach out by [filing an issue](https://github.com/coredns/coredns/issues).
+
+## Releases
+
+Releases will be created from the release branches. The release will follow the following convention as `<major>.<minor>.<patch>`. 
+All the major and minor releases will be made from the master. The tag will be of the format `v<major>.<minor>.<patch>`.
+
+After a minor release, a branch will be created with the format of `release-<major>.<minor>` from the minor tag. Once the branch is ready, all the patches will be released from that branch. 
```

Comment:
Yes, I agree this will create a lot more additional work and cherry-picking. But yes we can start creating a branches which are going to be in maintenance mode i.e.. (N-1).
```
N (1.10) branch = master
N-1 (1.9) branch = release-1.9
```